libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -1435,6 +1435,67 @@ impl NodeAPI for Greenlight {
         debug!("send_custom_message returned status {:?}", resp.status);
         Ok(())
     }
+
+    // Gets the routing hints related to all private channels that the node has
+    async fn get_routing_hints(&self) -> NodeResult<Vec<RouteHint>> {
+        let mut hints: Vec<RouteHint> = vec![];
+        let mut node_client = self.get_node_client().await?;
+        let peers = node_client
+            .list_peers(cln::ListpeersRequest::default())
+            .await?
+            .into_inner();
+
+        // For each peer look for a private channel and create a routing hint from it.
+        for peer in peers.peers {
+            if !peer.channels.is_empty() {
```

Comment:
IMO no need to check for this, because if the list is empty, the iterator below will just not iterate.

Reply:
Addressed